### PR TITLE
When AppCenter directory is not in "Assets" Unity can't build

### PR DESCRIPTION
**Pull request by AnastasiaKubova**
Things to consider before you submit the PR:

* [x] Has `CHANGELOG.md` been updated?
* [ ] ~Are tests passing locally?~
* [x] Are the files formatted correctly?
* [ ] ~Did you add unit tests if this modifies the Windows code?~
* [x] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Related PRs or issues

[AB#69372](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/69372)

https://github.com/microsoft/appcenter-sdk-unity/issues/321

### CODE REVIEW

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs [hunk 1]
```diff
@@ -3,12 +3,36 @@
...
+using System.Reflection;
+using System;
+using System.Runtime.CompilerServices;
 
 public class AppCenterSettingsContext : ScriptableObject
 {
-    public const string AppCenterPath = "Assets";
-    private const string SettingsPath = AppCenterPath + "/AppCenter/AppCenterSettings.asset";
-    private const string AdvancedSettingsPath = AppCenterPath + "/AppCenter/AppCenterSettingsAdvanced.asset";
+	private static readonly string SettingsPath = AppCenterPath + "/AppCenterSettings.asset";
```
Comment (annakocheshkova) [inline Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: .

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs [hunk 2]
```diff
@@ -3,12 +3,36 @@
...
+    private static readonly string AdvancedSettingsPath = AppCenterPath + "/AppCenterSettingsAdvanced.asset";
+    private static AppCenterSettingsContext Instance;
+
+    public static AppCenterSettingsContext GetInstance()
+    {
+        if(Instance == null)
```
Comment (annakocheshkova) [inline Assets/AppCenter/Editor/AppCenterSettingsContext.cs]:
```suggestion
        if (Instance == null)
```

**Review by annakocheshkova — DISMISSED** [review]

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs [hunk 3]
```diff
@@ -3,12 +3,35 @@
...
+    public static AppCenterSettingsContext GetInstance()
+    {
+        if (Instance == null)
+        {
+            Instance = new AppCenterSettingsContext();
+        }
+        return Instance;
+    }
+
+    private AppCenterSettingsContext()
```
Comment (annakocheshkova) [inline Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: image
https://forum.unity.com/threads/unityexception-scriptableobject-ctor-is-not-allowed-to-be-called-from-a-scriptableobject.466687/

**Review by annakocheshkova — CHANGES_REQUESTED** [review]

**Comment by annakocheshkova** [comment]
Also, I get `Failed to unload 'Assets/Plugins/AppCenter//AppCenterSettings.asset'`. This error is transient, but the double-slash is not.

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs (hunk 3 above)
Comment (olmisa) [reply Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Fixed.

Comment (olmisa) [reply Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Reworked

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs [hunk 4]
```diff
@@ -51,4 +84,5 @@ public static void DeleteSettingsInstanceAdvanced()
     {
         AssetDatabase.DeleteAsset(AdvancedSettingsPath);
     }
+
```
Comment (jaeklim) [inline Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Remove.

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs [hunk 5]
```diff
@@ -31,8 +44,28 @@ public static AppCenterSettingsAdvanced SettingsInstanceAdvanced
...
+        {
+            string temp;
+            for (int i = 0; i < folders.Length; i++)
+            {
+                temp = FindSubfolderPath(folders[i], searchFolder);
+                if (!string.IsNullOrEmpty(temp))
+                {
+                    return temp;
+                }
+
```
Comment (jaeklim) [inline Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Remove.

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,10 @@
 
 ## Release 2.6.0 (Under development)
 
+### App Center
+
+* **[Enhancement]** SDK now detects its location automatically.
```
Comment (jaeklim) [inline CHANGELOG.md]:
```suggestion
* **[Improvement]** Detect App Center SDK location automatically.
```

**Review by jaeklim — CHANGES_REQUESTED** [review]

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs (hunk 4 above)
Comment (olmisa) [reply Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Fixed

File: Assets/AppCenter/Editor/AppCenterSettingsContext.cs (hunk 5 above)
Comment (olmisa) [reply Assets/AppCenter/Editor/AppCenterSettingsContext.cs]: Fixed

File: CHANGELOG.md (hunk 6 above)
Comment (olmisa) [reply CHANGELOG.md]: Fixed

**Review by jaeklim — DISMISSED** [review]

**Review by guperrot — APPROVED** [review]

**Review by annakocheshkova — APPROVED** [review]

**Pull request merged**